src/device/pf_pdport.c

```diff
@@ -154,21 +187,40 @@ static int pf_pdport_save (pnet_t * net)
 
 /**
  * @internal
- * Get DAP subslot for using local port number
+ * Get DAP port subslot using local port number
+ *
  * @param net              InOut: The p-net stack instance
  * @param loc_port_num     In:    Local port number.
  *                                Valid range: 1 .. PNET_MAX_PORT
  * @return DAP subslot number for port identity
  */
 static uint16_t pf_pdport_loc_port_num_to_dap_subslot (int loc_port_num)
 {
-   CC_ASSERT (loc_port_num == PNET_PORT_1);
-   return PNET_SUBSLOT_DAP_INTERFACE_1_PORT_1_IDENT;
+   uint16_t subslot =
+      PNET_SUBSLOT_DAP_INTERFACE_1_PORT_1_IDENT + PNET_PORT_1 - loc_port_num;
+   return subslot;
+}
+
+static int pf_pdport_dap_subslot_to_local_port (uint16_t subslot)
```

Comment:
Add docstring.
Describe that 0 is returned for subslots out of range.

Reply:
Done